src/relation_engine_server/api_modules/api_v2.py

```diff
@@ -0,0 +1,61 @@
+import flask
+import jinja2
+
+from . import api_v1
+
+from ..utils import arango_client, spec_loader, auth, parse_json
+from ..exceptions import InvalidParameters
+
+
+def run_query():
+    """
+    Run a stored query from the spec repo.
+    Auth:
+     - only kbase re admins for ad-hoc queries
+     - public for views (views will have access controls within them based on params)
+    """
+    json_body = parse_json.get_json_body() or {}
+    if 'bind_vars' not in json_body:
+        json_body['bind_vars'] = {}
+    # Don't allow the user to set the special 'ws_ids' field
+    json_body['bind_vars']['ws_ids'] = []
+    auth_token = auth.get_auth_header()
+    # Fetch any authorized workspace IDs using a KBase auth token, if present
+    json_body['bind_vars']['ws_ids'] = auth.get_workspace_ids(auth_token)
+    # fetch number of documents to return
+    batch_size = int(flask.request.args.get('batch_size', 100))
+    if 'query' in json_body:
+        # Run an adhoc query for a sysadmin
+        auth.require_auth_token(roles=['RE_ADMIN'])
+        query_text = json_body['query']
+        resp_body = arango_client.run_query(query_text=query_text,
+                                            bind_vars=json_body['bind_vars'],
+                                            batch_size=batch_size)
+        return resp_body
+    if 'view' in flask.request.args:
+        # Run a query from a view name
+        view_name = flask.request.args['view']
+        view_str = spec_loader.get_view(view_name)
+        templ = jinja2.Environment(loader=jinja2.BaseLoader(), autoescape=True).from_string(view_str)
+        compiled_query = templ.render(**json_body.get('template_vars', {}))
```

Comment:
No super familiar with Jinja rendering but if there was an elegant way to determine which template variables were embedded (EX {{ bind_me }}) vs only used conditionally (EX {% if do_not_bind_me % }) we would not have to introduce this second template_vars class. (and maybe avoid going to a v2 api???)

Second thought: I think it might also be worth adding logic to check for @ws_ids in the aql before adding that call. It would remove the need for awkward unused variables in queries against public docs and make those call just a touch faster by cutting out the auth call

Reply:
Docs on writing jinja extensions: http://jinja.pocoo.org/docs/2.10/extensions/#module-jinja2.ext
I found it :fearful: 
There also may be some way to use a custom autoescape function. But I couldn't figure it out

I'm thinking now I want to look into other templating options that are more customizable.